lib/autoproj/workspace.rb

```diff
@@ -236,8 +236,19 @@ def overrides_dir
             File.join(config_dir, OVERRIDES_DIR)
         end
 
-        def load_config(reconfigure = false)
-            @config = Configuration.new(config_file_path)
+        # Load the configuration for this workspace from
+        # config_file_path
+        #
+        # @param [Boolean] reset Set to true to replace the configuration object,
+        #   set to false to load into the existing
+        # @return [Configuration] configuration object
+        def load_config(reconfigure = false, replace: true)
```

Comment:
Adding this optional behavior is going to be very fragile. From what I see, the "resetting the configuration" was there because everything in Configuration, so far, was saved to disk, and it was therefore easier.

Since it's not the case anymore, I'd suggest to simply change Configuration to make the reset unnecessary, for instance:

~~~
def reset
   @config.clear
   @modified = true
end
~~~

and then just change `load_config` to do a reset + load.

Reply:
Ack, that's the better approach

Reply:
Updated accordingly

Reply:
Great ! Thanks !

It seems that it broke a test in `test_workspaces.rb` though ... :P

Reply:
Assumptions of the test for flexmock on creating a new configuration were not true anymore, so updated the test.